src/operators.jl

```diff
@@ -442,3 +442,113 @@ This is equivalent to [`induced_subgraph`](@ref)`(g, neighborhood(g, v, d, dir=d
 with respect to `v` (i.e. `:in` or `:out`).
 """
 egonet(g::AbstractGraph, v::Integer, d::Integer; dir=:out) = g[neighborhood(g, v, d, dir=dir)]
+
+
+
+"""compute_shifts(n::Int, x::AbstractArray)
+
+Determines how many elements of vs are less than i for all i in 1:n.
+"""
+function compute_shifts(n::Integer, x::AbstractArray)
+    tmp = zeros(eltype(x), n)
+    tmp[x[2:end]] = 1
+    return cumsum!(tmp, tmp)
+end
+
+"""
+    merge_vertices(g::AbstractGraph, vs)
+
+Create a new graph where all vertices in vs have been aliased to the same vertex minimum(vs).
+"""
+function merge_vertices(g::AbstractGraph, vs)
+    labels = collect(1:nv(g))
+    # Use lowest value as new vertex id.
+    sort!(vs)
+    nvnew = nv(g) - length(@show unique(vs)) +1
+    v0 = minimum(vs)
+    v0 > 0 || error("minimum(vs) < 1")
+    maximum(vs) <= nv(g) || error("maximum(vs) > nv(g)")
+    for v in vs
+        labels[v] = v0
+    end
+    shifts = compute_shifts(nv(g), vs[2:end])
+    for v in vertices(g)
+        if labels[v] != v0
+            labels[v] -= shifts[v]
+        end
+    end
+
+    #if v in vs then labels[v] == v0 else labels[v] == v
+    newg = Graph(nvnew)
+    for e in edges(g)
+        u,w = src(e), dst(e)
+        if labels[u] != labels[w] #not a new self loop
+            add_edge!(newg, labels[u], labels[w])
+        end
+    end
+    return newg
+end
+
+
+"""
+    merge_vertices!(g, vs)
+
+Combine vertices specified in `vs` into single vertex whose
+index will be the lowest value in `vs`.
+
+Supports SimpleGraph.
+
+All edges running to vertices in `vs` connect to new
+merged vertex.
+
+
+Return a vector with new vertex values index by original vertex
+indices.
+"""
+function merge_vertices!(g::Graph, vs::Vector{T} where T <: Integer)
+    vs = sort!(unique(vs))
+    merged_vertex = shift!(vs)
+
+    x = zeros(Int, nv(g))
+    x[vs] = 1
+    new_vertex_ids = collect(1:nv(g)) .- cumsum(x)
+    new_vertex_ids[vs] = merged_vertex
+
+    for i in vertices(g)
+        # Adjust connections to merged vertices
+        if (i != merged_vertex) & !(i in vs)
+            nbrs_to_rewire = Set{eltype(g)}()
+            for j in out_neighbors(g, i)
+               if j in vs
```

Comment:
These are expensive unless you use a set. Since you're doing this in two places and you're not modifying `vs`, perhaps it makes sense to create a parallel set here, just for the membership tests?